shipyard_blueprints/fivetran/shipyard_fivetran/fivetran.py

```diff
@@ -1,33 +1,272 @@
+            method (str): The HTTP method of the request. Defaults to "GET".
+            payload (dict): The payload of the request. Defaults to None.
 
-    def determine_sync_status(self):
-        pass
+        Raises:
+            ExitCodeException: If the response status code is 401, 400, or others.
 
-    def update_connector(self):
-        pass
+        Returns:
+            dict: The JSON response from the Fivetran API
+        """
+        url = f"https://api.fivetran.com/v1/{endpoint}"
+        if payload:
+            resp = request(
+                method,
+                url,
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json;version=2",
+                },
+                json=payload,
+                auth=self.auth,
+            )
+        else:
+            resp = request(
+                method,
+                url,
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json;version=2",
+                },
+                auth=self.auth,
+            )
+        if resp.ok:
+            return resp.json()
+        self.logger.error(f"Error: {resp.status_code} - {resp.text}")
+        if resp.status_code == 401:
+            raise ExitCodeException(
+                f'{resp.json().get("message")}', self.EXIT_CODE_INVALID_CREDENTIALS
+            )
+        elif resp.json().get("code") == "NotFound_Integration":
+            raise ExitCodeException(
+                f'{resp.json().get("message")}', self.EXIT_CODE_SYNC_INVALID_SOURCE_ID
+            )
+        elif resp.status_code == 400:
+            raise ExitCodeException(
+                f'{resp.json().get("message")}', self.EXIT_CODE_BAD_REQUEST
+            )
+        else:
+            raise ExitCodeException(f"{resp.json()}", self.EXIT_CODE_UNKNOWN_ERROR)
 
-    def connect(self) -> int:
-        """ Connect to Fivetran API
+    def trigger_sync(
+        self,
+        connector_id: str,
+        force: bool = True,
+        wait_for_completion: bool = False,
+        poke_interval: int = 30,
+    ):
+        """
+        Trigger a sync on a specific connector.
+
+        Parameters:
+            connector_id (str): The ID of the connector
+            force (bool): Whether to force the sync. Defaults to True.
+            wait_for_completion (bool): Whether to wait for the sync to complete. Defaults to False.
+            poke_interval (int): Interval in seconds to check for sync completion. Defaults to 30.
+
+        Raises:
+            ExitCodeException: If an error occurs while triggering sync.
+        """
+        if wait_for_completion:
+            self.logger.info(
+                "Getting connection details for last successful and failed syncs..."
+            )
+            prev_success, prev_failure = self._get_latest_success_and_failure(
+                connector_id
+            )
```

Comment:
Any reason why you wouldn't put the block from below right here since you are handling the `wait_for_completion` check? 

```python
                new_success, new_failure = prev_success, prev_failure
                while prev_success == new_success and prev_failure == new_failure:
                    self.logger.info("Waiting for sync to complete...")
                    time.sleep(poke_interval)
                    new_success, new_failure = self._get_latest_success_and_failure(
                        connector_id
                    )
                self.logger.info("Sync completed")
                self.logger.info("Checking for new failure")
                if (
                    prev_failure
                    and new_failure != prev_failure
                    or not prev_failure
                    and new_failure
                ):
                    raise ExitCodeException(
                        f"Sync failed at {new_failure}",
                        self.EXIT_CODE_SYNC_REFRESH_ERROR,
                    )
                else:
                    self.logger.info("No new failure detected")

```

Reply:
I wanted a snapshot of the values before the sync was attempted in the unlikely scenario there is an edge case that can mark the succeeded_at or failed at almost instantly